proposals/handler.go

```diff
@@ -262,6 +262,11 @@ func (h *Handler) handleProposal(ctx context.Context, expHash types.Hash32, peer
 		preGenesis.Inc()
 		return fmt.Errorf("proposal before effective genesis: layer %v", p.Layer)
 	}
+	if p.Layer < h.clock.CurrentLayer() {
+		// old proposals have no use for the node
```

Comment:
in this case the enclosed ballot should be fetched via sync, if not already.

Reply:
another option is to save the ballot only

Reply:
what do you think about maintaining variable that is updated by block generator after it is done with this layer?

defaults to current layer on startup, but it will allow to stop accepting proposals earlier

Reply:
it's a good idea. i chose to use `mesh.ProcessedLayer()` to be the threshold for the following reasons
- it's updated every time a block is generated, optimistically or not
- hare can terminate out of order, so block can be generated out of order as well in the non-optimistic case
- proposal handler already have mesh interface ready to use